srml/babe/src/lib.rs

```diff
@@ -262,11 +285,17 @@ impl<T: Trait + staking::Trait + Duration> session::OneSessionHandler<T::Account
 
 		// What was the next epoch is now the current epoch
 		let randomness = Self::randomness_change_epoch(epoch_index);
+
+		EpochStartSlot::mutate(|previous| *previous += <T::EpochDuration>::get());
```

Comment:
This should saturate.

Reply:
Also, should account for skipping of at least one whole epoch as discussed offline. So add `T::EpochDuration::get()` in a loop until `now - previous < T::EpochDuration::get()`